File: compositor/src/main/java/org/westford/compositor/core/events/RenderOutputDestroyed.java
package org.westford.compositor.core.events;

import com.google.auto.value.AutoValue;
import org.westford.compositor.core.RenderOutput;
import org.westford.compositor.protocol.WlOutput;

import javax.annotation.Nonnull;

@AutoValue
public abstract class RenderOutputDestroyed {
    public static RenderOutputDestroyed create(@Nonnull final WlOutput wlOutput) {
        return new AutoValue_RenderOutputDestroyed(wlOutput);
    }

    @Nonnull
    public abstract WlOutput getWlOutput();
}

File: compositor/src/main/kotlin/org/westford/compositor/core/events/RenderOutputDestroyed.kt
package org.westford.compositor.core.events

import com.google.auto.value.AutoValue
import org.westford.compositor.protocol.WlOutput

@AutoValue abstract class RenderOutputDestroyed {

    abstract val wlOutput: WlOutput

    companion object {
        fun create(wlOutput: WlOutput): RenderOutputDestroyed {
            return AutoValue_RenderOutputDestroyed(wlOutput)
        }
    }
}
